homeassistant/components/velux/config_flow.py

```diff
@@ -43,13 +54,54 @@ async def async_step_user(
                 errors["base"] = "unknown"
             else:
                 return self.async_create_entry(
-                    title=user_input[CONF_HOST],
+                    title=user_input[CONF_NAME] or user_input[CONF_HOST],
                     data=user_input,
                 )
 
-        data_schema = self.add_suggested_values_to_schema(DATA_SCHEMA, user_input)
+        data_schema = self.discovery_schema or self.add_suggested_values_to_schema(
+            DATA_SCHEMA, user_input
+        )
         return self.async_show_form(
             step_id="user",
             data_schema=data_schema,
             errors=errors,
         )
+
+    async def async_step_dhcp(
+        self, discovery_info: DhcpServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle discovery by DHCP."""
+        return await self._async_handle_discovery(
+            {
+                CONF_HOST: discovery_info.ip,
+                CONF_MAC: format_mac(discovery_info.macaddress),
+                CONF_NAME: discovery_info.hostname.upper().replace("LAN_", ""),
+            }
+        )
+
+    async def _async_handle_discovery(
+        self, discovery_info: dict[str, Any]
+    ) -> ConfigFlowResult:
+        """Prepare configuration for a discovered Velux device."""
+        # The hostname ends with the last 4 digits of the device MAC address.
+        await self.async_set_unique_id(discovery_info[CONF_NAME])
+
+        self._abort_if_unique_id_configured(
+            updates={CONF_HOST: discovery_info[CONF_HOST]}
+        )
+
+        # Abort if config_entry already exists without unigue_id configured.
+        for entry in self.hass.config_entries.async_entries(DOMAIN):
+            if (
+                entry.data[CONF_HOST] == discovery_info[CONF_HOST]
+                and entry.unique_id is None
+            ):
```

Comment:
Never mind, I guess this also works.

Be sure that we also need to check if the entry is loaded, this way we know that the velux device we are talking about is actually working

Reply:
I added the loaded check.